src/common/misc.c

```diff
@@ -769,3 +772,78 @@ daos_hlc2timestamp(uint64_t hlc, time_t *ts)
 	*ts = tspec.tv_sec;
 	return 0;
 }
+
+#define MORE_ENV_VARS 200
+/**
+ * Constructor to pre-allocate an env vars pointers array enough big to
+ * avoid re-allocation during any setenv() (in DAOS and any 3rd party
+ * lib) and thus prevent any race when using getenv().
+ */
+static __attribute__((constructor)) void
+daos_grow_env_array(void)
```

Comment:
I wonder if the problem here is what I saw online

https://www.reddit.com/r/cpp_questions/comments/10oyuzi/attribute_constructor_shared_library_not_getting/

There may be no server side callers to daos_sgl_init on the server side.